_posts/2022-01-10-reassign-partitions.md

```diff
@@ -0,0 +1,539 @@
+  (To utilise a newly added JBOD volume it's probably more convenient to perform a global rebalance using `KafkaRebalance` once the volume has been added).
+
+In the rest of this post we'll dive into how the tool works, and then go through some examples based on these use cases.
+
+## Tool modes
+
+The `bin/kafka-reassign-partitions.sh` tool allows you to reassign partitions to different brokers or disks.
+
+The tool has two fundamental modes of operation:
+
+- `--execute`: This initiates a ressignment that you describe using a JSON file. We'll refer to this file throughout as reassignment.json, though you can name it how you like. Note that when the execution of an --execute command completes the reassignment has only been started.
+- `--verify`: This checks whether reassignment that you previously started (using --execute) has actually completed.
+
+Constructing a reassignment.json can be anything from a bit tedious to a piece of work in its own right. To help with this the tool provides an optional third mode, `--generate`, to generate one from a list of topics you provide in a different JSON file, which we'll be calling topics.json. You can use the reassignment.json that is produced by a --generate invocation directly, or modify it further prior to actually making changes using --execute.
+
+## Partition Reassignment in detail
+
+When you make changes to a partition's assignment to brokers there are three possible cases:
+
+1. Changes which require moving data between brokers.
+2. Changes which require moving data between volumes on _the same_ broker.
+3. Change which require no data movement at all.
+
+The first case leads to additional inter-broker network traffic, in addition to the normal traffic required for replication.
+To avoid overloading the cluster, it is recommended to always set a throttle rate to limit the bandwidth used by the reassignment.
+This can be done using the `--throttle` option in the `--execute` model.
+This sets the maximum allowed bandwidth in bytes per second, for example `kafka-reassign-partitions.sh --execute --throttle 5000000 ...` would set the limit to 5 MB/s.
+
+Throttling might cause the reassignment to take longer to complete.
+
+* If the throttle is too low, the newly assigned brokers will not be able to keep up with records being published and the reassignment will never complete.
+
+* If the throttle is too high, the overall health of the cluster may be impacted.
+
+You can use the `kafka.server:type=FetcherLagMetrics,name=ConsumerLag,clientId=([-.\w]+),topic=([-.\w]+),partition=([0-9]+)` metric (which with the default `spec.kafka.metrics` settings in your Kafka CR would be named in Prometheus as `kafka_server-fetcherlagmetrics_consumerlag`) to observe how far the followers are lagging behind the leader. You can refer to this [example](https://github.com/strimzi/strimzi-kafka-operator/blob/main/packaging/examples/metrics/kafka-metrics.yaml) for configuring metrics
```

Comment:
`which with the default `spec.kafka.metrics` settings in your Kafka CR would be named in Prometheus as `kafka_server-fetcherlagmetrics_consumerlag`` ... it is not the default for the field. It is how our metrics examples are configured.